rapyuta_io_sdk_v2/client.py

```diff
@@ -187,3 +187,1012 @@ def get_project(self, project_guid: str = None) -> Munch:
+
+        v2api_host = self.config.hosts.get("v2api_host")
+
+        response = self.c.get(
+            url=f"{v2api_host}/v2/deployments/",
+            headers=headers,
+        )
+
+        handle_server_errors(response)
+
+        return munchify(response.json())
+
+    def create_deployment(self, project_guid: str, body: dict) -> Munch:
+        """Create a new deployment.
+
+        Returns:
+            Munch: Deployment details as a Munch object.
+        """
+        headers = self.config.get_headers()
+        if project_guid:
+            headers["project"] = project_guid
+
+        v2api_host = self.config.hosts.get("v2api_host")
+
+        response = self.c.post(
+            url=f"{v2api_host}/v2/deployments/",
+            headers=headers,
+            json=body,
+        )
+
+        handle_server_errors(response)
+
+        return munchify(response.json())
+
+    def get_deployment(self, project_guid: str, name: str) -> Munch:
+        """Get a deployment by its name.
+
+        Args:
+            name (str): Deployment name
+
+        Returns:
+            Munch: Deployment details as a Munch object.
+        """
+        headers = self.config.get_headers()
+        if project_guid:
+            headers["project"] = project_guid
+
+        v2api_host = self.config.hosts.get("v2api_host")
+
+        response = self.c.get(
+            url=f"{v2api_host}/v2/deployments/{name}/",
+            headers=headers,
+        )
+
+        handle_server_errors(response)
+
+        return munchify(response.json())
+
+    def update_deployment(self, project_guid: str, name: str, body: dict) -> Munch:
+        """Update a deployment by its name.
+
+        Args:
+            name (str): Deployment name
+            body (dict): Update details
+
+        Returns:
+            Munch: Deployment details as a Munch object.
+        """
+        headers = self.config.get_headers()
+        if project_guid:
+            headers["project"] = project_guid
+
+        v2api_host = self.config.hosts.get("v2api_host")
+
+        response = self.c.put(
+            url=f"{v2api_host}/v2/deployments/{name}/",
+            headers=headers,
+            json=body,
+        )
+
+        handle_server_errors(response)
+
+        return munchify(response.json())
+
+    def delete_deployment(self, project_guid: str, name: str) -> Munch:
+        """Delete a deployment by its name.
+
+        Args:
+            name (str): Deployment name
+
+        Returns:
+            Munch: Deployment details as a Munch object.
+        """
+        headers = self.config.get_headers()
+        if project_guid:
+            headers["project"] = project_guid
+
+        v2api_host = self.config.hosts.get("v2api_host")
+
+        response = self.c.delete(
+            url=f"{v2api_host}/v2/deployments/{name}/",
+            headers=headers,
+        )
+
+        handle_server_errors(response)
+
+        return munchify(response.json())
+
+    def poll_deployment(
```

Comment:
thought: Maybe we can consider moving this outside the client into a generic poll_until function that can accept any rapyuta.io object and implement poll. This is also not a direct API call so shouldn't be part of the client.